src/test/regress/sql/multi_alter_table_add_constraints.sql

```diff
@@ -0,0 +1,387 @@
+--
+-- MULTI_ALTER__TABLE_ADD_CONSTRAINTS
+--
+-- Test checks whether constraints of distributed tables can be adjusted using
+-- the ALTER TABLE ... ADD CONSTRAINT ... command.
+
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1400000;
+ALTER SEQUENCE pg_catalog.pg_dist_jobid_seq RESTART 1400000;
+
+-- Check "UNIQUE CONSTRAINT"
+CREATE TABLE unique_test_table(id int, name varchar(20));
+SELECT create_distributed_table('unique_test_table', 'id');
+
+-- Can only add unique constraint on distribution column (or group 
+-- of columns including distribution column)
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_name UNIQUE(name);
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id UNIQUE(id);
+
+-- Error out from the related shard. Table can not have two rows with same id.
+INSERT INTO unique_test_table VALUES(1, 'Ahmet');
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+
+ALTER TABLE unique_test_table DROP CONSTRAINT unn_id;
+
+-- Can create unique constraint over multiple columns including
+-- distribution column
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id UNIQUE(id);
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id_name UNIQUE(id, name);
+
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+DROP TABLE unique_test_table;
+
+-- Check "UNIQUE CONSTRAINT" with reference table
+CREATE TABLE unique_test_table_ref(id int, name varchar(20));
+SELECT create_reference_table('unique_test_table_ref');
+
+-- We can add unique constraint on any column with reference tables
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_name UNIQUE(name);
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_id UNIQUE(id);
+
+-- Error out. Since the table can not have two rows with the same id. 
+INSERT INTO unique_test_table_ref VALUES(1, 'Ahmet');
+INSERT INTO unique_test_table_ref VALUES(1, 'Mehmet');
+
+-- We can add unique constraint with multiple columns 
+ALTER TABLE unique_test_table_ref DROP CONSTRAINT unn_id;
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_id_name UNIQUE(id,name);
+INSERT INTO unique_test_table_ref VALUES(1, 'Mehmet');
+
+DROP TABLE unique_test_table_ref;
+
+-- Check "UNIQUE CONSTRAINT" with append table
+CREATE TABLE unique_test_table_append(id int, name varchar(20));
+SELECT create_distributed_table('unique_test_table_append', 'id', 'append');
+
+-- We can add unique constraint on any column with reference tables
+ALTER TABLE unique_test_table_append ADD CONSTRAINT unn_name UNIQUE(name);
```

Comment:
We should prevent it. Append-distributed tables may have overlapping shards and typically do.